### PR TITLE
Remove -_ from sIds

### PR DESCRIPTION
**Pull request by philipperolet**
Description
---
Following this [discussion](https://dust4ai.slack.com/archives/C050SM8NSPK/p1719299080025739)

This PR moves sIds to a 62-length charset [A-Za-z0-9]

Note: since 62 is not a power of 2, we'll always have to discard some bytes. This PR handles it by retrying in unlikely cases where this causes an issue.

Risk & deploy
---
na
not necessary to backfill IMO

### CODE REVIEW

File: front/lib/utils.ts [hunk 1]
```diff
@@ -291,5 +303,5 @@ function alphanumFromByte(byte: number) {
   if (index < 62) {
     return CHAR_0 + index - 52;
   }
-  return index === 62 ? 45 : 95;
+  return 0;
```
Comment (spolu) [inline front/lib/utils.ts:306]: Null characters are preserved in the string representation even if they don't count for the length no?

Comment (spolu) [reply front/lib/utils.ts:306]: They actually count for the length as well

```
> new Buffer([62,0,0,64]).toString()
'>\x00\x00@'
> new Buffer([62,0,0,64]).toString().length
4
```

Comment (philipperolet) [reply front/lib/utils.ts:306]: :see_no_evil: will fix

Comment (philipperolet) [reply front/lib/utils.ts:307]: Could either do a theoretically better code with a carry param & a reduce, or just filter the null string
Went for the second that is IMHO easier to follow

**Comment by philipperolet** [comment]
After thinking, base conversion seems overkill: converting from base 256 to base 62 will involve 256^i and 62^j computations with i/j going up to 10, and I don't see any simple way to do this while avoiding actually performing arithmetic on those large numbers (62 = 2* 31 which is prime)

General algorithm does not seem trivial actually: https://cs.stackexchange.com/questions/21736/time-complexity-of-base-conversion

Will think of something else

**Comment by philipperolet** [comment]
After wrapping my head, solutions:
1. stick to the skip (original code with null string)
2. Math.random(0,62) when the modulo is 62 or 63 (~ 1 call to random per generation)
3. Ignore the issue (just have a bit bias towards A-F chars)

Other options exist but seem overkill. I realize this is already overthought. No real preference between the 4 above; defaulting to solution 2 which I sense cleaner, updating the PR in consequence

**Comment by philipperolet** [comment]
PMRR

**Review by spolu — APPROVED** [review]
Sorry for missing this! LGTM

**Pull request merged**